lumicks/pylake/kymo.py

```diff
@@ -189,10 +189,13 @@ def _plot(self, image, **kwargs):
         width_um = self._ordered_axes()[0]["scan width (um)"]
         ts = self.timestamps
         duration = (ts[0, -1] - ts[0, 0]) / 1e9
+        dt = (ts[1, 0] - ts[0, 0]) / 1e9
+        linetime = ts.shape[0] * dt
```

Comment:
Naively, I would have expected `ts[0, 1] - ts[0, 0]`. Is this different?

Reply:
Yes, that worked fine for the 1D case, but when I expanded the test out to 2D data the times were still slightly offset from the center of the pixels. 

I believe the issue comes down to the fact that the `extent` argument is setting the outer edges whereas `ts[0, 1] - ts[0, 0` gives you the number of bins. The middle panel below is the result if you calculate the line time how you suggest

![image](https://user-images.githubusercontent.com/61475504/101155845-02e3b000-3628-11eb-9540-46d9a54880d8.png)

Reply:
That's weird. I observe the opposite?

```python
plt.close('all')
plt.figure(figsize=(7, 7))

def plot_test(method):
    row_time = 5
    t = np.array([[0, 1, 2], 
                  np.array([0, 1, 2]) + row_time, 
                  np.array([0, 1, 2]) + 2 * row_time, 
                  np.array([0, 1, 2]) + 3 * row_time]).T

    duration = t[0, -1] - t[0, 0]
    width_um = 2

    line_time = t[0, 1] - t[0, 0] if method == 0 else (t[1, 0] - t[0, 0]) * t.shape[0]
    title = 't[0, 1] - t[0, 0]' if method == 0 else '(t[1, 0] - t[0, 0]) * shape[0]'
    plt.title(title)

    plt.imshow(t)
    kwargs = dict(
        extent=[-0.5 * line_time, duration + 0.5 * line_time, width_um, 0],
        aspect=(t.shape[0] / t.shape[1]) * (duration / width_um)
    )
    plt.imshow(t, **kwargs)
    
plt.subplot(2, 1, 1)
plot_test(0)
plt.subplot(2, 1, 2)
plot_test(1)
plt.tight_layout()```

![image](https://user-images.githubusercontent.com/19836026/101162906-3e37ac00-3633-11eb-8cf3-8ef6b284e168.png)

Reply:
You're right. I wasn't taking into account the flyin time between lines in my simple test data, and the discrepancy for the real data (large dataset) was minimal so it looked ok. Fixed now

Reply:
👍